EcosiaTests/Analytics/AnalyticsSpyTests.swift

```diff
@@ -33,8 +39,13 @@ final class AnalyticsSpy: Analytics {
     }
 
     var menuClickItemCalled: Analytics.Label.Menu?
+    var menuClickExpectation: XCTestExpectation?
     override func menuClick(_ item: Analytics.Label.Menu) {
         menuClickItemCalled = item
+        DispatchQueue.main.async {
+            self.menuClickExpectation?.fulfill()
+            self.menuClickExpectation = nil // Prevent multiple fulfillments
+        }
```

Comment:
Thank you for handling the flakyness of the menu tests, it's indeed quite tricky 🙏 

Can you elaborate why setting this to nil is needed? Shouldn't we expect this to be called only once per test?

Reply:
Tbh there wasn't much of a need for them. Once I solved the problem at the root I should have removed them. Did it now. Thanks! 😉